flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/CodeGenUtils.scala

```diff
@@ -1020,16 +1021,19 @@ object CodeGenUtils {
     }
 
     // convert internal format to target type
-    val externalResultTerm = if (isInternalClass(targetDataType)) {
-      s"($targetTypeTerm) ${internalExpr.resultTerm}"
+    val (externalResultTerm, resultType) = if (isInternalClass(targetDataType)) {
```

Comment:
can we use more meaningful names here? `resultType` does not improve code readability. `externalResultTypeTerm` is the right name here.